src/src/tx_main.cpp

```diff
@@ -433,19 +467,11 @@ void ICACHE_RAM_ATTR SendRCdataToRF()
 {
   uint32_t now = millis();
   static uint8_t syncSlot;
-  uint32_t SyncInterval;
-  bool skipSync;
 
-  if (firmwareOptions.no_sync_on_arm)
-  {
-    SyncInterval = 250;
-    skipSync = IsArmed() || InBindingMode;
-  }
-  else
-  {
-    SyncInterval = (connectionState == connected) ? ExpressLRS_currAirRate_RFperfParams->SyncPktIntervalConnected : ExpressLRS_currAirRate_RFperfParams->SyncPktIntervalDisconnected;
-    skipSync = InBindingMode;
-  }
+  uint32_t SyncInterval = (connectionState == connected) ? ExpressLRS_currAirRate_RFperfParams->SyncPktIntervalConnected : ExpressLRS_currAirRate_RFperfParams->SyncPktIntervalDisconnected;
```

Comment:
In 'Race' mode it can take up to 5s for a sync packet to be sent to the rx before the tlm rate is changes.  Then another ~1s for the required number of missed tlm packets.  This feel like a long time, particularly if its the beginning of a race.  Adding an extra condition for SyncInterval can help significantly speed this up.  And I dont see a detriment to maintaining a high sync rate when disarmed.

`uint32_t SyncInterval = (connectionState == connected && config.GetTlm() != TLM_RATIO_DISARMED) `

Reply:
You make it sound like the connection is severely crippled for those 6 seconds, but we're talking about 4 packets per second. That's less than 1% of packets, which is nothing. Don't you arm and then "go on the time in less than 5", so those packets wouldn't even be used anyway because you're sitting on the blocks? I thought it was better to just let it happen naturally over adding more code because the benefit is never going to be perceptible. You know me, I'm always pushing for less specialized code when it doesn't make a significant difference.

Reply:
Sitting on the block or ground your lq isn't the perfect.  Then out of the other five quads 1 or 2 will have tlm on for some reason, causing more dropped packets.  5s can easily turn into 10s with a missed sync packet.

Might as well have it done quickly.

Reply:
That is true, and I think your solution is pretty clever too since the number of sync packets in that mode doesn't matter when disarmed and it is all still right in the same place in the code. Added.